databuilder/databuilder/extractor/dashboard/apache_superset/__init__.py

```diff
@@ -0,0 +1,174 @@
+import abc
+from functools import reduce
+from typing import (
+    Any, Dict, Generator, Iterator, List, Tuple,
+)
+
+import requests
+from bs4 import BeautifulSoup
+from dateutil import parser
+from pyhocon import ConfigFactory, ConfigTree
+
+from databuilder.extractor.base_extractor import Extractor
+
+type_fields_mapping = List[Tuple[str, str, Any, Any]]
+
+
+class ApacheSupersetBaseExtractor(Extractor):
+    """
+    Base class to create extractors pulling any dashboard metadata from Apache Superset.
+    """
+    APACHE_SUPERSET_PROTOCOL = 'apache_superset_protocol'
+    APACHE_SUPERSET_HOST = 'apache_superset_host'
+    APACHE_SUPERSET_PORT = 'apache_superset_port'
+    APACHE_SUPERSET_USER = 'apache_superset_user'
+    APACHE_SUPERSET_PASSWORD = 'apache_superset_password'
+    APACHE_SUPERSET_PAGE_SIZE = 'apache_superset_page_size'
+    APACHE_SUPERSET_EXTRACT_PUBLISHED_ONLY = 'apache_superset_extract_published_only'
+
+    DASHBOARD_GROUP_NAME = 'dashboard_group_name'
+    DASHBOARD_GROUP_ID = 'dashboard_group_id'
+    DASHBOARD_GROUP_DESCRIPTION = 'dashboard_group_description'
+
+    PRODUCT = 'product'
+    CLUSTER = 'cluster'
+
+    DRIVER_TO_DATABASE_MAPPING = 'driver_to_database_mapping'
+
+    DEFAULT_DRIVER_TO_DATABASE_MAPPING = {
+        'postgresql': 'postgres',
```

Comment:
what is this for?

Reply:
we need to render table uri list for DashboardTable model. table uri could be postgres://gold.db_name/table_name . there is no field in database model in superset to directly pull this info so I am relying on sqlalchemy conn uri (this one is available and required for database model). but this uri more focuses on driver (like `mysql+pymysql`) rather than technology (`mysql` in that case). the same goes for postgres (`postgresql` in sql conn uri where amundsen extractors rely on `postgres` name).  this is the configuration option to allow users mapping between drivers they use and underlying technologies.